test/widgets/message_list_test.dart

```diff
@@ -484,6 +487,22 @@ void main() {
 
       debugNetworkImageHttpClientProvider = null;
     });
+
+    testWidgets('Bot user is distinguished by showing an icon', (tester) async {
+      prepareBoringImageHttpClient();
+
+      final users = List.generate(3, (index) => eg.user(isBot: index == 0));
```

Comment:
This is probably clearer as a list literal with three explicit `eg.user` calls: `[eg.user(isBot: true), eg.user(isBot: false),` etc. Not much longer that way, and less logic for the reader to unpack.